### PR TITLE
Add support for K8s Pod Security Policies (PSP)

### PR DESCRIPTION
**Pull request by almaslennikov**
Enable auto creation of PodSecurityPolicies to apply to all operator Pods

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -100,3 +100,6 @@ issues:
     - text: "Magic number: 1"
       linters:
         - gomnd
+    - path: api/v1alpha1/zz_generated.deepcopy.go
```
Comment (moshe010) [inline .golangci.yml:103]: why this was added?

File: manifests/stage-whereabouts-cni/0020-cluster_role.yaml [hunk 2]
```diff
@@ -28,3 +28,13 @@ rules:
...
+  {{- if .RuntimeSpec.PspEnabled }}
+- apiGroups:
+    - policy
+  resources:
+    - podsecuritypolicies
+  verbs:
+    - use
+  resourceNames:
+    - network-operator-privileged
+  {{- end }}
```
Comment (moshe010) [inline manifests/stage-whereabouts-cni/0020-cluster_role.yaml:40]: missing new line

File: .golangci.yml (hunk 1 above)
Comment (almaslennikov) [reply .golangci.yml:104]: Added an explaining comment

File: api/v1alpha1/nicclusterpolicy_types.go [hunk 3]
```diff
@@ -101,6 +101,14 @@ type SecondaryNetworkSpec struct {
 	IpamPlugin *ImageSpec `json:"ipamPlugin,omitempty"`
 }
 
+// PSPSpec describes configuration for PodSecurityPolicies to apply for all Pods
+type PSPSpec struct {
+	// Enabled indicates if PodSecurityPolicies needs to be enabled for all Pods
+	// +optional
+	// +kubebuilder:default:={}
```
Comment (e0ne) [inline api/v1alpha1/nicclusterpolicy_types.go:108]: Why default is an empty dict?

**Pull request merged**